zap/src/main/java/org/parosproxy/paros/model/Session.java

```diff
@@ -287,6 +289,8 @@ public void run() {
 
     protected void open(String fileName) throws DatabaseException, IOException, Exception {
 
+        final CountDownLatch latch = new CountDownLatch(2);
```

Comment:
Please, provide information about the issues that you are trying to fix. This is deadlocking in daemon and command line modes.

Reply:
In my tests this didn't cause a deadlock. There 2 InvokeAndWait under the open() method in the Session.java.
These were not loading, potentially because of the exception from the setcharset. 
InvokeLater allows both areas to run while waiting for them to finish prior to moving onto loading more of the session.

Reply:
I understand what the code is doing not what is trying to fix. If this is being done as workaround for the other issue then it shouldn't be needed if the other issue is fixed.

To reproduce the deadlock you just need to load a session in daemon and command line modes (you can also check the code, the count down only happens when there's view).

Reply:
Oh, ic what happened. I will fix this.